zipline/pipeline/loaders/blaze/_core.pyx

```diff
@@ -131,6 +131,61 @@ ctypedef fused AsArrayKind:
     AsBaselineArray
 
 
+# This is the core algorithm for formatting raw blaze data into the baseline +
+# adjustments format required for consumption by the Pipeline API.
+#
+# Logically, we think of each row of the input data as representing a single
+# event. Each event carries four pieces of information:
+#
+#   asof_date - The date on which the event occurred.
+#   timestamp - The date on which we learned about the event.
+#   sid       - The sid to which the event pertains.
+#   value     - The value of the column being updated by the event.
+#
+# The essential idea of this algorithm is to process events in timestamp-sorted
+# order, updating our worldview as each new event arrives. This models how we
+# would actually process events in real time.
+#
+# When we process a new event, we first check if the event should be processed:
+#
+# - We skip events pertaining to sids that weren't requested.
+# - We skip events whose timestamp/as_of are after all the dates we're
```

Comment:
might be worth being explicit: `whose timestamp or as_of`. I found myself wondering if you want `and` or `or`>